tests/test_azure.py

```diff
@@ -35,3 +56,39 @@ def test_parse_image_name(monkeypatch):
     }
 
     assert az.parse_image_name('do not match') is None
+
+
+def test_cleanup_sle_images_container_too_many(monkeypatch):
+    test_name = 'quantity'
+    monkeypatch.setattr(Azure, 'check_credentials', lambda *args, **kwargs: True)
+    monkeypatch.setattr(BlockBlobService, 'list_blobs', list_blobs_mock)
+    monkeypatch.setattr(BlockBlobService, 'delete_blob', delete_blob_mock)
+    monkeypatch.setattr(Provider, 'cfgGet', mock_cfgGet)
+    az = Azure('fake')
+    az.cleanup_sle_images_container(BlockBlobService(account_name='openqa', account_key='www'), test_name)
+    deleted = ['SLES15-SP2-Azure-HPC.x86_64-0.9.0-Build1.43.vhd', 'SLES15-SP2-Azure-HPC.x86_64-0.9.1-Build1.3.vhd',
+               'SLES15-SP2-BYOS.x86_64-0.9.3-Azure-Build2.36.vhd', 'SLES15-SP2-BYOS.x86_64-0.9.6-Azure-Build1.3.vhd']
+    for item in deleted:
+        assert item in delete_calls[test_name]
```

Comment:
```suggestion
    assert delete_calls[test_name] == deleted
```
The order must fit with the algorithm but you also check with this, that we do not delete to much.